json-rpc/integration-tests/src/lib.rs

```diff
@@ -1730,6 +1730,65 @@ impl PublicUsageTest for GetEventByVersionWithProofTest {
     }
 }
 
+pub struct GetResourcesTest;
+impl Test for GetResourcesTest {
+    fn name(&self) -> &'static str {
+        "jsonrpc::get-resource-test"
+    }
+}
+
+impl PublicUsageTest for GetResourcesTest {
+    fn run<'t>(&self, ctx: &mut PublicUsageContext<'t>) -> Result<()> {
+        let env = JsonRpcTestHelper::new(ctx.url().to_owned());
+        let address = format!("{:x}", diem_sdk::types::account_config::diem_root_address());
+        let response = env.send("get_resources", json!([address]));
+
+        let value = response.result.unwrap();
+        let resources = value.as_object().unwrap();
+        // check that the CurrencyInfo<XDX> resource looks ok as a sanity check
+        let currency_info = resources
+            .get("0x1::Diem::CurrencyInfo<0x1::XDX::XDX>")
+            .unwrap();
+        assert_eq!(
+            currency_info,
+            &json!({
+              "total_value": 0,
+              "preburn_value": 0,
+              "to_xdx_exchange_rate": {
+                "value": 4294967296_u64
+              },
+              "is_synthetic": true,
+              "scaling_factor": 1000000,
+              "fractional_part": 1000,
+              "currency_code": "XDX",
+              "can_mint": false,
+              "mint_events": {
+                "counter": 0,
+                "guid": "0x0a000000000000000000000000000000000000000a550c18"
```

Comment:
although it's nice to see this format more consistent with other blockchain like Ethereum, but it seems we are creating a new format of hex-encoded string, as other APIs return hex-encoded string without "0x".

Reply:
I probably would like to start to use "0x" prefix for hex-encoded data in new API, but in json-rpc API, it maybe better to keep it consistent with other apis. Otherwise, the guid extracted here, probably can't be used in get_events api

Reply:
That makes sense; I will change this. I wanted to have a prefix to distinguish hex strings from utf8 strings, but since we're not already doing this it doesn't make sense to start now. Perhaps worth thinking about in the REST API, though.